agent/status.go

```diff
@@ -76,11 +76,20 @@ func nodeToSystemStatus(status pb.NodeStatus_Type) pb.SystemStatus_Type {
 	}
 }
 
-func isDegraded(status pb.SystemStatus) bool {
+func isDegraded(resp *pb.StatusResponse) bool {
+	status := resp.GetStatus()
 	switch status.Status {
 	case pb.SystemStatus_Unknown, pb.SystemStatus_Degraded:
 		return true
 	}
+	for _, node := range status.Nodes {
+		for _, probe := range node.Probes {
+			if probe.Status == pb.Probe_Temporary {
+				// Consider temporary failures as degrading the cluster state
+				return true
```

Comment:
Because as I wrote in description, only the `healthz` endpoint is sensitive to warnings - otherwise, only critical errors cause the system status to go `degraded`. it's something we talked about with @klizhentas.

Reply:
I reverted these changes as they seem to break symmetry of `status` and healthz results.